cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/PatConfig.java

```diff
@@ -45,19 +48,34 @@ public String getPasswordName() {
 
   @Nullable
   public String getUsername() {
-    return username;
+    // If username is not specified, use the default username.
+    return username == null || username.isEmpty() ? DEFAULT_USERNAME : username;
   }
 
   /**
    * Validate PatConfig.
-
+   *
    * @return a collection of {@link RepositoryValidationFailure}.
    */
-  public Collection<RepositoryValidationFailure> validate() {
+  public Collection<RepositoryValidationFailure> validate(Provider provider) {
     Collection<RepositoryValidationFailure> failures = new ArrayList<>();
     if (passwordName == null || passwordName.equals("")) {
-      failures.add(new RepositoryValidationFailure("'passwordName' must be specified in 'patConfig'."));
+      failures.add(
+          new RepositoryValidationFailure("'passwordName' must be specified in 'patConfig'."));
     }
+    if (getUsername() == null) {
+      failures.add(
+          new RepositoryValidationFailure("'username' must be specified in 'patConfig'."));
+    }
+
+    if (provider == Provider.BITBUCKET_CLOUD || provider == Provider.BITBUCKET_SERVER) {
```

Comment:
Added the check here as wanted to isolate all PAT related validation at one place. Con of this approach is the provider details is getting passed to PAT config. Better alternative can be to create separate PAT class for each provider but I think that is unneeded complexity ATM as the provider list is finite and very small.